ARGV: add support for std::env::args()

alastairreid opened the pull request:
This PR extends cargo-verify to be able to handle command line arguments, documents how it works and adds a simple example and regression test.

Commit:
ARGV: add support for std::env::args()

fshaked commented on demos/simple/argv/src/main.rs:
```diff
@@ -0,0 +1,25 @@
+/// to Rust functions when verifying them.
+///
+/// Should pass:
+///    `cargo verify . foo`
```
'cargo run' passes all the arguments after '--' to the program. Can you make 'cargo verify' behave the same?

Commit:
Docs: use -- in argv examples

Also tweaks order of flags in cargo-verify examples so that
the directory is always the last argument (before any -- args).

And update the CHANGELOG file.
